cmd/csi-node-driver-registrar/main.go

```diff
@@ -66,9 +67,10 @@ var (
 	pluginRegistrationPath  = flag.String("plugin-registration-path", "/registration", "Path to Kubernetes plugin registration directory.")
 	kubeletRegistrationPath = flag.String("kubelet-registration-path", "", "Path of the CSI driver socket on the Kubernetes host machine.")
 	healthzPort             = flag.Int("health-port", 0, "(deprecated) TCP port for healthz requests. Set to 0 to disable the healthz server. Only one of `--health-port` and `--http-endpoint` can be set.")
-	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including the health check indicating whether the registration socket exists, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--health-port` and `--http-endpoint` can be set.")
+	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including pprof and the health check indicating whether the registration socket exists, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--health-port` and `--http-endpoint` can be set.")
```

Comment:
@verult it looks like if you set httpEndpoint, it always enables the health check. Would we potentially want to enable the health check and pprof independently? Or maybe it doesn't matter since pprof is for debugging anyway.

Reply:
pprof can still be disabled by a separate flag, so the only case this prevents is having pprof on but health check off. I think it's OK not to allow that case since the health check is fairly lightweight. If we need to allow that case we could add a flag in the future.